portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -106,4 +128,31 @@ private int getNumofComments(HttpServletRequest request) {
 
     return numOfComments;
   }
+
+  private String getUploadedFileUrl(HttpServletRequest request, String formInputElementName) {
+        Map<String, List<BlobKey>> blobs = blobstore.getUploads(request);
+        List<BlobKey> blobKeys = blobs.get(formInputElementName);
+
+        if (blobKeys == null || blobKeys.isEmpty()) {
+            return "";
+        }
+
+        BlobKey blobKey = blobKeys.get(0);
```

Comment:
Is it possible for blobKeys to contain more than one element? If so, how would you determine which one to use?

I'm not very familiar with the Blobstore API, so maybe there is a guarantee that the first key in the list is always the one you want, and it would make sense that in a simple application like this you can guarantee that you're only putting one element in there.

This is sort of just a long way of me saying this should be fine for here, but maybe at least add a comment that says why you are using the first element in the list and what value you expect it to have.

Reply:
I'm going to be honest, I don't really know. I took a look at the documentation and the tutorials again, but I can't seem to find why they use the first element of the list. Maybe I just haven't looked hard enough.

Reply:
Ah, interesting that that isn't explained better then. The code you have here is fine, especially if it's the convention for the library. This is just something that I'd keep an eye on in the future.